go/vt/vtgate/gateway/discoverygateway.go

```diff
@@ -109,6 +109,16 @@ func createDiscoveryGateway(hc discovery.HealthCheck, serv srvtopo.Server, cell
 	// We set sendDownEvents=true because it's required by TabletStatsCache.
 	hc.SetListener(dg, true /* sendDownEvents */)
 
+	// Resolve the cell to region mapping at initialization time before
+	// loading any of the tablets.
+	if topoServer != nil {
+		log.Infof("loading region information for cells: %v", *cellsToWatch)
+		err := topo.InitRegions(context.Background(), topoServer, strings.Split(*cellsToWatch, ","))
```

Comment:
nit -I think it won't hurt to use a context with a timeout here. Even though this is only initialize, I would prefer to see a timeout error and enter in a crash loop on start, than getting  blocked in this call forever.